android/KMEA/app/src/main/java/com/tavultesoft/kmea/data/CloudDataJsonUtil.java

```diff
@@ -220,4 +237,187 @@ public static File getLexicalModelCacheFile(Context context) {
+            } else {
+              fontSource = removeJsonObjectAtIndex(fontSource, i);
+            }
+          }
+        }
+
+        if (updateJsonFont) {
+          JSONArray copy = fontSource;
+          jsonFont.remove(KMManager.KMKey_FontSource);
+          jsonFont.put(KMManager.KMKey_FontSource, copy);
+        }
+      }
+    } catch (JSONException e) {
+      Log.e(TAG, "findTTF exception" + e);
+    }
+  }
+
+  // Parse the fontSource JSONArray to see if it contains a .ttf font
+  private static boolean hasTTFFont(JSONArray fontSource) {
+    try {
+      for (int i = 0; i < fontSource.length(); i++) {
+        String s = fontSource.getString(i);
+        if (FileUtils.isTTFFont(s)) {
+          return true;
+        }
+      }
+      return false;
+    } catch (JSONException e) {
+      Log.e(TAG, "hasTTFFont exception" + e);
+      return false;
+    }
+  }
+
+  // From https://stackoverflow.com/questions/27427999/remove-jsonobeject-before-android-api-lvl-19
+  private static JSONArray removeJsonObjectAtIndex(JSONArray source, int index) throws JSONException {
+    if (index < 0 || index > source.length() - 1) {
+      throw new IndexOutOfBoundsException();
+    }
+
+    final JSONArray copy = new JSONArray();
+    for (int i=0, count = source.length(); i<count; i++) {
+      if (i != index) {
+        copy.put(source.get(i));
+      }
+    }
+    return copy;
+  }
+
+  public static ArrayList<String> fontUrls(JSONObject jsonFont, String baseUri, boolean isOskFont) {
+    if (jsonFont == null)
+      return null;
+
+    ArrayList<String> urls = new ArrayList<String>();
+    JSONArray fontSource = jsonFont.optJSONArray(KMManager.KMKey_FontSource);
+    if (fontSource != null) {
+      int fcCount = fontSource.length();
+      for (int i = 0; i < fcCount; i++) {
+        String fontSourceString;
+        try {
+          fontSourceString = fontSource.getString(i);
+
+          if (FileUtils.hasFontExtension(fontSourceString)) {
+            urls.add(baseUri + fontSourceString);
+          } else if (isOskFont && FileUtils.hasSVGViewBox(fontSourceString)) {
+            String fontFilename = FileUtils.getSVGFilename(fontSourceString);
+            urls.add(baseUri + fontFilename);
+          }
+        } catch (JSONException e) {
+          return null;
+        }
+      }
+    } else {
+      String fontSourceString;
+      try {
+        fontSourceString = jsonFont.getString(KMManager.KMKey_FontSource);
+        if (FileUtils.hasFontExtension(fontSourceString)) {
+          urls.add(baseUri + fontSourceString);
+        } else if (isOskFont && FileUtils.hasSVGViewBox(fontSourceString)) {
+          String fontFilename = FileUtils.getSVGFilename(fontSourceString);
+          urls.add(baseUri + fontFilename);
+        }
+      } catch (JSONException e) {
+        return null;
+      }
+    }
+
+    return urls;
+  }
+
+  public static JSONObject findMatchingKeyBoardByID(JSONArray aKeyboards, String aKbId)
```

Comment:
Another one to rename: `findMatchingKeyBoardByID`. Might as well replace all instances of `KeyBoard` with `Keyboard`.